xarray/tests/test_cftimeindex.py

```diff
@@ -132,16 +134,26 @@ def date_dict(
     list(ISO8601_LIKE_STRING_TESTS.values()),
     ids=list(ISO8601_LIKE_STRING_TESTS.keys()),
 )
-def test_parse_iso8601_like(string, expected):
-    result = parse_iso8601_like(string)
+@pytest.mark.parametrize("five", [False, True], ids=["4Y", "5Y"])
+@pytest.mark.parametrize("sign", ["", "+", "-"], ids=["None", "plus", "minus"])
+def test_parse_iso8601_like(five, sign, string, expected):
+    pre = "1" if five else ""
+    datestring = sign + pre + string
+    result = parse_iso8601_like(datestring)
+    expected = expected.copy()
+    expected.update(year=sign + pre + expected["year"])
     assert result == expected
 
-    if result["microsecond"] is None:
+    # check malformed single digit addendum
+    # tests for year/month/day excluded as year can be 4 or 5 digits
```

Comment:
i couldn't follow this comment.

Reply:
I've rephrased the text, main issue is for a single added digit the string will be interpreted as having a 5-digit year.